src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java

```diff
@@ -0,0 +1,179 @@
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.AppInsights.DRAFT_STORE;
+import static uk.gov.hmcts.ccd.domain.model.draft.DraftResponseBuilder.aDraftResponse;
+
+
+@Service
+@Qualifier(DefaultDraftGateway.QUALIFIER)
+public class DefaultDraftGateway implements DraftGateway {
+
+    public static final String QUALIFIER = "default";
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultDraftGateway.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final String DRAFT_ENCRYPTION_KEY_HEADER = "Secret";
+    private static final int RESOURCE_NOT_FOUND = 404;
+    public static final String DRAFT_STORE_DOWN_ERR_MESSAGE = "The draft service is currently down, please refresh your browser or try again later";
+    private static final String RESOURCE_NOT_FOUND_MSG = "No draft found ( draft reference = '%s' )";
+    private static final String DRAFT_STORE_DESERIALIZATION_ERR_MESSAGE = "Unable to read from draft service";
+
+    @Qualifier("draftsRestTemplate")
+    @Autowired
+    private final RestTemplate restTemplate;
+    private final SecurityUtils securityUtils;
+    private final ApplicationParams applicationParams;
+    private final AppInsights appInsights;
+
+    @Inject
+    public DefaultDraftGateway(
+        final RestTemplate restTemplate,
+        final SecurityUtils securityUtils,
+        final ApplicationParams applicationParams,
+        final AppInsights appInsights) {
+        this.restTemplate = restTemplate;
+        this.securityUtils = securityUtils;
+        this.applicationParams = applicationParams;
+        this.appInsights = appInsights;
+    }
+
+    @Override
+    public Long save(final CreateCaseDraftRequest draft) {
+        try {
+            HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.add(DRAFT_ENCRYPTION_KEY_HEADER, applicationParams.getDraftEncryptionKey());
+            final HttpEntity requestEntity = new HttpEntity(draft, headers);
+            final Instant start = Instant.now();
+            HttpHeaders responseHeaders = restTemplate.exchange(applicationParams.draftBaseURL(),
+                                                                HttpMethod.POST,
+                                                                requestEntity,
+                                                                HttpEntity.class).getHeaders();
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "Create", duration.toMillis(), true);
+            return getDraftId(responseHeaders);
+        } catch (Exception e) {
+            LOG.warn("Error while saving draft", e);
+            throw new ServiceException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
```

Comment:
this call could fail for many reasons, not only draft store down. Shall we make this error message more generic?

Reply:
This is as expected. Per design a generic message is supposed to be returned regardless of the reason.